### ISSUE TITLE
Geolocation.watchPosition stops emitting on single error

### ISSUE THREAD

**Issue by Barryrowe** [driftyco/ionic-native#533]
Plugin: Geolocation (cordova-plugin-geolocation)

Issue:
The way that the error callback is bound in the Geolocation.watchPosition method prevents the observable from mimicking the Geolocation API. 

When using the native events:
```
navigator.geolocation.watchPosition((success) => console.log("success"), (error) => console.log("error"), options);
```
The watch does not stop emitting successes when an error occurs. 

With the ```observer.error``` being bound to the error callback of the geolocation API, the returned observable is completed and will no longer emit position updates when the api becomes available again after a single error.

I have submitted PR #532 to switch this to emit the error values out to the Observable as well.  This would let a consumer use ```Observable.partition``` or ```Observable.filter``` or any solution they wish to handle success and error callbacks from the geolocation.watchPosition API results.  I also updated the docs to show an example of using ```filter``` for filtering out error results.

**Comment by Barryrowe** [comment]
In the meantime, this can be worked around by editing your ```node_modules/ionic-native/dist/plugins/geolocation.js``` changing line 63 to match:

```
var watchId = navigator.geolocation.watchPosition(observer.next.bind(observer), observer.next.bind(observer), options);
```

**Comment by ihadeed** [comment]
PR merged and will be published in v1.3.21 soon
